### PR TITLE
[FEATURE] Document canonical API

### PR DESCRIPTION
**Pull request by DanielSiepmann**
Allow developers and integrators to find ways to alter generation of
canonical URL.

### CODE REVIEW

**Review by susannemoog — APPROVED** [review]

**Comment by haassie** [comment]
I would love to mention that it is using the configuration for cacheHash generation. If an URL is worth caching (because it has different content) it is worth having a canonical as well. So options like `$GLOBALS['TYPO3_CONF_VARS']['FE']['cacheHash']['excludedParameters']` and `$GLOBALS['TYPO3_CONF_VARS']['FE']['cacheHash']['cachedParametersWhiteList']` are useful for canonicals as well.

**Review by haassie — CHANGES_REQUESTED** [review]
See previous comment

**Comment by DanielSiepmann** [comment]
I'll extend the PR.

**Review by haassie — APPROVED** [review]

File: Documentation/ApiOverview/CanonicalApi/Index.rst [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+   $GLOBALS['TYPO3_CONF_VARS']['FE']['additionalCanonicalizedUrlParameters'][] = 'example_argument_name[second_level]';
+
+Arguments in general should be excluded from cHash as well as ``additionalCanonicalizedUrlParameters``.
+See the possible options in :ref:`caching`, regarding excluding arguments from cHash.
+
+The idea behind that is:
+
+If a URL is worth caching (because it has different content) it is worth having a canonical as well.
+
+   — https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/1326#issuecomment-788741312
```
Comment (linawolf) [inline Documentation/ApiOverview/CanonicalApi/Index.rst:35]: Why is there an uncommented url to an issue comment in git here? It should at least have a link text And be a propper link. And then again, why link to the issue?

**Review by linawolf — CHANGES_REQUESTED** [review]

File: Documentation/ApiOverview/CanonicalApi/Index.rst (hunk 1 above)
Comment (DanielSiepmann) [reply Documentation/ApiOverview/CanonicalApi/Index.rst:35]: Its not uncommented, it was the source of the cite, until the cite was unintended and therefore no longer is a cite.

https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/pull/1326/commits/f497b53f485ee7fda4fa5af5f107182763ec7b95 still had this as a cite. I'm citing Richard.

Comment (linawolf) [reply Documentation/ApiOverview/CanonicalApi/Index.rst:35]: Sorry about that, I reverted my change

**Review by linawolf — APPROVED** [review]

**Pull request merged**